cmd/juno/juno.go

```diff
@@ -207,8 +228,24 @@ func NewCmd(config *node.Config, run func(*cobra.Command, []string) error) *cobr
 
 		// TextUnmarshallerHookFunc allows us to unmarshal values that satisfy the
 		// encoding.TextUnmarshaller interface (see the LogLevel type for an example).
-		return v.Unmarshal(config, viper.DecodeHook(mapstructure.ComposeDecodeHookFunc(
-			mapstructure.TextUnmarshallerHookFunc(), mapstructure.StringToTimeDurationHookFunc())))
+		if err := v.Unmarshal(config, viper.DecodeHook(mapstructure.ComposeDecodeHookFunc(
+			mapstructure.TextUnmarshallerHookFunc(), mapstructure.StringToTimeDurationHookFunc()))); err != nil {
+			return err
+		}
+
+		// Set custom network
+		if v.GetString(cnNameF) != "" {
+			config.Network = utils.Network{}
+			config.Network.Name = v.GetString(cnNameF)
+			config.Network.FeederURL = v.GetString(cnFeederURLF)
+			config.Network.GatewayURL = v.GetString(cnGatewayURLF)
+			config.Network.L1ChainID = new(big.Int).SetInt64(v.GetInt64(cnL1ChainIDF))
+			config.Network.L2ChainID = v.GetString(cnL2ChainIDF)
+			coreContractAddress := common.HexToAddress(v.GetString(cnCoreContractAddressF))
+			config.Network.CoreContractAddress = coreContractAddress
+		}
```

Comment:
We need to set the metainfo here right? Otherwise block hash will panic. Or we can check if it is `nil` during the block hash calculation.

Reply:
I think requiring users to set metaInfo for custom networks probably adds unnecessary complexity. I've updated the code so it shouldn't panic if meta info isn't set.

Reply:
We could set the default ourselves so the user doesn't have to specify.

```go
network := utils.Network{
    // ... 
    BlockHashMetaInfo: &utils.BlockHashMetaInfo{
        // ...
    },
}
```

Reply:
Sure, that works too. I've set the First07Block as 0, fallBackSeqAddress the same as Goerli (and all non-mannet networks), and let UnverifRange unset.